File: Data Structures & Algorithms/products-of-array-discluding-self/submission-9.java
class Solution {
    public int[] productExceptSelf(int[] nums) {

        int[] solution = new int[nums.length];

        int running = nums[0];

        for(int i = 0; i < solution.length; i++) {
            solution[i] = 1;
        }

        for(int i = 1; i < nums.length; i++) {
            solution[i] = running;
            running *= nums[i];
        }

        running = nums[nums.length - 1];

        for(int i = nums.length - 2; i >= 0; i--) {
            solution[i] *= running;
            running *= nums[i];
        }

        return solution;
        
    }
}  

File: Data Structures & Algorithms/top-k-elements-in-list/submission-9.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:

        freq = {}

        for n in nums:
            freq[n] = freq.get(n, 0) + 1

        freq_list = [[] for _ in range(len(nums) + 1)]

        for key, value in freq.items():

            freq_list[value].append(key)

        res = []

        for num_list in reversed(freq_list):
            
            for n in num_list:
                res.append(n)

                if len(res) == k: return res
            
        return res
            


        








        